pilot/pkg/serviceregistry/kube/controller/controller_test.go

```diff
@@ -670,7 +670,7 @@ func TestGetProxyServiceInstancesWithMultiIPsAndTargetPorts(t *testing.T) {
 					TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: 8080},
 				},
 			},
-			wantNum: 2,
+			wantNum: 1,
```

Comment:
Might be worth adding higher level tests that validate the listeners/clusters are right. Maybe in sidecar_simulation_test

Reply:
I tried to, but simulation tests are all based on service entry, which does not suffer from this dedup logic.